sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Listeners/PollLogsStrategy.cs

```diff
@@ -76,10 +80,28 @@ public async Task RegisterAsync(BlobServiceClient blobServiceClient, BlobContain
 
             containerRegistrations.Add(triggerExecutor);
 
-            if (!_logListeners.ContainsKey(blobServiceClient))
+            if (targetBlobServiceClient == default)
             {
-                BlobLogListener logListener = await BlobLogListener.CreateAsync(blobServiceClient, _logger, cancellationToken).ConfigureAwait(false);
-                _logListeners.Add(blobServiceClient, logListener);
+                // If no target client is specified, use the primary.
```

Comment:
I don't see why we need this special handling for if a target isn't specified. We're already taking a BlobContainerClient that is coming from the "target" account. Ultimately that's coming from when the trigger binding was created [here](https://github.com/Azure/azure-sdk-for-net/blob/a5d95e75b679c0ce0b6e218ed99ea8aa539f39e1/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Triggers/BlobTriggerBinding.cs#L195). The "data" client is the "target" client. I think we can simplify here, particularly based on my other comment not to even pass the "primary" account down.

Reply:
Removed this